datadog_checks_dev/datadog_checks/dev/tooling/commands/meta/manifest.py

```diff
@@ -0,0 +1,203 @@
+# (C) Datadog, Inc. 2021-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import json
+
+import click
+
+from datadog_checks.dev.fs import write_file
+from datadog_checks.dev.tooling.datastructures import JSONDict
+from datadog_checks.dev.tooling.utils import get_manifest_file, get_valid_integrations, load_manifest
+
+from ..console import CONTEXT_SETTINGS, abort, echo_info, echo_success
+
+# This means the value is either not present in the old manifest, or there's logic needed to compute it
+SKIP_IF_FOUND = "SKIP"
+
+# Static text to let users know these values need updating by hand
+TODO_FILL_IN = "TODO Please Fill In"
+
+# Validate what manifest versions we can support and what we can upgrade from->to
+SUPPORTED_MANIFEST_VERSIONS = ["1.0.0", "2.0.0"]
+SUPPORTED_VERSION_UPGRADE_PATHS = {"1.0.0": ["2.0.0"]}
+
+# JSONDict map all v2 fields to their v1 counterparts
+# Skipping any fields that need manual intervention of custom logic
+V2_TO_V1_MAP = JSONDict(
+    {
+        "/manifest_version": SKIP_IF_FOUND,
+        "/app_id": "/integration_id",
+        "/classifier_tags": [],
+        "/display_on_public_website": "/is_public",
+        "/tile": {},
+        "/tile/overview": "README.md#Overview",
+        "/tile/configuration": "README.md#Setup",
+        "/tile/support": "README.md#Support",
+        "/tile/changelog": "CHANGELOG.md",
+        "/tile/description": "/short_description",
+        "/tile/title": "/public_title",
+        "/tile/media": [],
+        "/author": {},
+        "/author/homepage": "/author/homepage",
+        "/author/name": "/author/name",
+        "/author/support_email": "/maintainer",
+        "/oauth": {},
+        "/assets": {},
+        "/assets/integration": {},
```

Comment:
We also need to add a field for "/assets/integration/id"

Reply:
I believe we decided that we'd remove integration_id from the manifest and instead have `integration sync` auto add this field and just have it be exactly `app_id`

Reply:
Ah, in that case we need to update the online validator.

Reply:
I added a backlog card for getting that updated